python/cudf/cudf/dataframe/series.py

```diff
@@ -1574,7 +1574,10 @@ def one_hot_encoding(self, cats, dtype="float64"):
         A sequence of new series for each category.  Its length is determined
         by the length of ``cats``.
         """
-
+        if isinstance(cats, Series):
+            cats = cats.to_pandas()
+        else:
+            cats = pd.Series(cats)
```

Comment:
I guess we don't control what a user passes in for `cats` huh?

Reply:
Yes, made this check more generic.

Reply:
yes